app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -19,6 +19,10 @@ def handle_second_factor_locked_user
   end
 
   def check_already_authenticated
+    unless user_session
+      redirect_to new_user_session_path
```

Comment:
yeah that would be totally fine. What I was suggesting above was pretty much that but via a shared example group for minimal spec code duplication. eg https://github.com/18F/identity-idp/blob/master/spec/support/shared_examples_for_password_validation.rb

You'd need to pass a param for the actual controller call, though, so maybe more work than it's worth https://www.relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples